lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
We definitely need one of `data`, `links`, or `meta` according to the spec. We definitely don't have the first (remember empty data is incorrect here). Empty `links` also seems invalid, so that's a no go. [The section on meta](http://jsonapi.org/format/#document-meta) seems to indicate empty meta is valid. So If a better alternative exists, I don't know it.